src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/PackageDetailsTabViewModel.cs

```diff
@@ -0,0 +1,106 @@
+                SetAndRaisePropertyChanged(ref _selectedTab, value);
+            }
+        }
+
+        public PackageDetailsTabViewModel()
+        {
+            _readmeTabEnabled = true;
+            Tabs = new ObservableCollection<RenderedViewModelBase>();
+        }
+
+        public PackageMetadataTab GetSelectedTab()
+        {
+            return SelectedTab is DetailControlModel ? PackageMetadataTab.PackageDetails : PackageMetadataTab.Readme;
+        }
+
+        public async Task InitializeAsync(DetailControlModel detailControlModel, INuGetPackageFileService nugetPackageFileService, ItemFilter currentFilter, PackageMetadataTab initialSelectedTab)
+        {
+            var nuGetFeatureFlagService = await ServiceLocator.GetComponentModelServiceAsync<INuGetFeatureFlagService>();
+            _readmeTabEnabled = await nuGetFeatureFlagService.IsFeatureEnabledAsync(NuGetFeatureFlagConstants.RenderReadmeInPMUI);
+
+            ReadmePreviewViewModel = new ReadmePreviewViewModel(nugetPackageFileService, currentFilter);
+            ReadmePreviewViewModel.IsVisible = _readmeTabEnabled;
+            DetailControlModel = detailControlModel;
+
+            Tabs.Add(ReadmePreviewViewModel);
+            Tabs.Add(DetailControlModel);
+
+            SelectedTab = _readmeTabEnabled && initialSelectedTab == PackageMetadataTab.Readme
+                ? ReadmePreviewViewModel
+                : DetailControlModel;
+
+            DetailControlModel.PropertyChanged += DetailControlModel_PropertyChanged;
+            ReadmePreviewViewModel.PropertyChanged += ReadMePreviewViewModel_PropertyChanged;
+        }
+
+        public async Task SetCurrentFilterAsync(ItemFilter filter)
+        {
+            if (_readmeTabEnabled)
+            {
+                await ReadmePreviewViewModel.SetCurrentFilterAsync(filter);
+            }
+        }
+
+        public void Dispose()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+            _disposed = true;
+            DetailControlModel.PropertyChanged -= DetailControlModel_PropertyChanged;
+        }
+
+        private void ReadMePreviewViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (ReadmePreviewViewModel is not null
+                && e.PropertyName == nameof(ReadmePreviewViewModel.IsVisible))
+            {
+                if (!ReadmePreviewViewModel.IsVisible)
+                {
+                    SelectedTab = DetailControlModel;
+                }
+            }
+        }
+
+        private void DetailControlModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            ThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                if (_readmeTabEnabled)
+                {
+                    await ReadmePreviewViewModel.SetPackageMetadataAsync(DetailControlModel.PackageMetadata, CancellationToken.None);
+                }
+            });
+        }
```

Comment:
Another way of writing this would be to have the PackageDetailsTabViewModel let any listeners know when the PackageMetadata changed so the listener (in this case the ReadmePreviewViewModel) can update its own PackageMetadata. This removes the responsibility of this ViewModel updating the ReadmePreviewViewModel directly.